src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -96,6 +95,7 @@
      * only for single-sample VCFs.
      */
     protected String remappedSampleName = null;
+    protected static boolean vcfIndexUseStringsCache = "true".equals(System.getProperty("VCF_INDEX_USE_STRINGS_CACHE"));
```

Comment:
I think that the property name is rather confusing, this has nothing to do with indexes. it affects the entire vcf codec system, which is why this PR potentially has very wide-reaching effects and should not be taken lightly.